core/Azure.Mcp.Core/src/Services/ProcessExecution/ExternalProcessService.cs

```diff
@@ -26,130 +241,475 @@ public async Task<ProcessResult> ExecuteAsync(
+                oce.Data["ProcessExitCheckException"] = exitCheck.CheckException;
+                _logger.LogWarning(exitCheck.CheckException, "Could not determine process exit state during cancellation for command: {Command}", command);
+                break;
+
+            default:
+                throw new InvalidOperationException($"Unexpected exit status: {exitCheck.Status}");
+        }
+
+        oce.Data["ProcessName"] = process.SafeName();
+        oce.Data["ProcessId"] = process.SafeId();
+        oce.Data["ProcessExitStatus"] = exitCheck.Status.ToString();
+
+        // we deliberately preserve and rethrow (via 'throw;') the original OCE at call site.
+    }
+
+    /// <summary>
+    /// Reads either stdout or stderr from a <see cref="Process"/> asynchronously.
+    /// Handlers are attached in the constructor, and reading begins when
+    /// <see cref="StartReading"/> is called after the process has started.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Intended usage pattern:
+    /// </para>
+    /// <list type="number">
+    ///   <item><description>Create and configure the <see cref="Process"/> with redirected streams.</description></item>
+    ///   <item><description>Construct <see cref="ProcessStreamReader"/> (handlers attach immediately).</description></item>
+    ///   <item><description>Start the process.</description></item>
+    ///   <item><description>Call <see cref="StartReading"/> to begin event-driven reading.</description></item>
+    ///   <item><description>Await the returned task to obtain the full stream content.</description></item>
+    /// </list>
+    ///
+    /// <para>
+    /// The reader does not handle cancellation directly - the underlying <see cref="Process"/>
+    /// owns the reader threads, which naturally terminate when the process exits or is killed.
+    /// This type:
+    /// </para>
+    /// <list type="bullet">
+    ///   <item><description>Accumulates received lines into a buffer.</description></item>
+    ///   <item><description>Completes when <c>e.Data == null</c>, the signal that the stream closed.</description></item>
+    ///   <item><description>
+    ///     Includes a safety-net completion in <see cref="Dispose"/> so callers cannot hang if
+    ///     a close event is never raised (only relevant if used outside <c>ExecuteAsync</c>).
+    ///   </description></item>
+    /// </list>
+    /// </remarks>
+    private sealed class ProcessStreamReader : IDisposable
+    {
+        private readonly Process process;
+        private readonly bool isErrorStream;
+        private readonly ILogger<ExternalProcessService> _logger;
+        private readonly StringBuilder buffer = new();
+        private readonly TaskCompletionSource<string> tcs = new(TaskCreationOptions.RunContinuationsAsynchronously);
+        private readonly DataReceivedEventHandler handler;
+        private bool readingStarted;
+        private bool disposed;
```

Comment:
The field naming needs to be consistent and match .editorconfig style rules:

```suggestion
        private readonly Process _process;
        private readonly bool _isErrorStream;
        private readonly ILogger<ExternalProcessService> _logger;
        private readonly StringBuilder _buffer = new();
        private readonly TaskCompletionSource<string> _tcs = new(TaskCreationOptions.RunContinuationsAsynchronously);
        private readonly DataReceivedEventHandler _handler;
        private bool _readingStarted;
        private bool _disposed;
```